alpaca/broker/enums.py

```diff
@@ -0,0 +1,317 @@
+    USER_CONFIGURATIONS = "trading_configurations"
+
+
+class ClearingBroker(str, Enum):
+    """
+    An enum for representing what Clearing broker an Account is assigned to
+    """
+
+    Apex = "APEX"
+    ETC = "ETC"
+    IC = "IC"
+    Velox = "VELOX"
+    Vision = "VISION"
+    Self = "SELF"
+
+
+class CIPProvider(str, Enum):
+    """
+    Enum representing what CIP provider was used.
+
+    see https://alpaca.markets/docs/api-references/broker-api/accounts/accounts/#cip-provider for more info
+    """
+
+    ALLOY = "alloy"
+    TRULIOO = "trulioo"
+    ONFIDO = "onfido"
+    VERIFF = "veriff"
+    JUMIO = "jumio"
+    GETMATI = "getmati"
+
+
+class CIPStatus(str, Enum):
+    """
+    An enum representing the status of the CIPInfo
+
+    see https://alpaca.markets/docs/api-references/broker-api/accounts/accounts/#cip-status for more info
+    """
+
+    COMPLETE = "complete"
+    WITHDRAWN = "withdrawn"
+
+
+class CIPResult(str, Enum):
+    """
+    see https://alpaca.markets/docs/api-references/broker-api/accounts/accounts/#cip-result for more info
+    """
+
+    CLEAR = "clear"
+    CONSIDER = "consider"
+
+
+class CIPApprovalStatus(str, Enum):
+    """
+    Either `approved` or `rejected`
+    """
+
+    APPROVED = "approved"
+    REJECTED = "rejected"
+
+
+class ActivityType(str, Enum):
+    """
+    Represents what kind of Activity an instance of TradeActivity or NonTradeActivity is.
+
+    Please see https://alpaca.markets/docs/api-references/broker-api/accounts/account-activities/#enumactivitytype
+    for descriptions of each of the types
+    """
+
+    FILL = "FILL"
+    ACATC = "ACATC"
+    ACATS = "ACATS"
+    CIL = "CIL"
+    CSD = "CSD"
+    CSW = "CSW"
+    DIV = "DIV"
+    DIVCGL = "DIVCGL"
+    DIVCGS = "DIVCGS"
+    DIVNRA = "DIVNRA"
+    DIVROC = "DIVROC"
+    DIVTXEX = "DIVTXEX"
+    FEE = "FEE"
+    INT = "INT"
+    JNLC = "JNLC"
+    JNLS = "JNLS"
+    MA = "MA"
+    PTC = "PTC"
+    REORG = "REORG"
+    SPIN = "SPIN"
+    SPLIT = "SPLIT"
+
+    def is_trade_activity(self) -> bool:
+        """
+        A simple check to see if the ActivityType represents a type that belongs to TradeActivity's.
+
+        Currently, the check is just against FILL. However, this might change in the future so we are adding this helper
+        func here to help ease against future changes.
+
+        Returns:
+            bool: returns true if this ActivityType represents a TradeActivity
+        """
+
+        return self.value == self.FILL
+
+
+class TradeActivityType(str, Enum):
+    """
+    Represents the type of TradeActivity.
+
+    Please see https://alpaca.markets/docs/api-references/broker-api/accounts/account-activities/#attributes
+    """
+
+    PARTIAL_FILL = "partial_fill"
+    FILL = "fill"
+
+
+class OrderSide(str, Enum):
+    """
+    Represents what side this order was executed on.
+    """
+
+    BUY = "buy"
+    SELL = "sell"
+
+
+class OrderStatus(str, Enum):
+    NEW = "new"
```

Comment:
Should we add a docstring here?

Reply:
Yes, this model is only temporary here anywho as it should probably be moved to the common module. I don't think broker has any unique order status' ? :thinking: